detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/BracesOnIfStatementsSpec.kt

```diff
@@ -0,0 +1,301 @@
+        }
+
+        @Test
+        fun `reports inconsistent block without braces`() {
+            val findings = subject.compileAndLint(BLOCK_WITHOUT_BRACES_FAIL)
+
+            assertThat(findings).hasSize(8)
+            assertThat(findings).hasTextLocations(
+                119 to 123,
+                185 to 189,
+                278 to 282,
+                365 to 367,
+                479 to 483,
+                577 to 581,
+                655 to 657,
+                815 to 819
+            )
+        }
+
+        @Test
+        fun `reports inconsistent block with braces`() {
+            val findings = subject.compileAndLint(BLOCK_WITH_BRACES_FAIL)
+
+            assertThat(findings).hasSize(9)
+            assertThat(findings).hasTextLocations(
+                129 to 133,
+                201 to 205,
+                254 to 256,
+                321 to 325,
+                469 to 473,
+                576 to 580,
+                644 to 648,
+                770 to 774,
+                867 to 869
+            )
+        }
+
+        @Test
+        fun `reports if when else is multi-statement`() {
+            val findings = subject.compileAndLint(
+                """
+                    fun f() {
+                        if (true) println() else { println(); println() }
+                    }
+                """.trimIndent()
+            )
+            assertThat(findings).hasSize(1)
+            assertThat(findings).hasTextLocations(14 to 16)
+        }
+    }
+
+    @Nested
+    inner class `braces always` {
+        val subject = createSubject("always", "always")
+
+        @Test
+        fun `does not report correct block with braces`() {
+            val findings = subject.compileAndLint(BLOCK_WITH_BRACES_PASS)
+
+            assertThat(findings).isEmpty()
+        }
+
+        @Test
+        fun `reports incorrect block with braces`() {
+            val findings = subject.compileAndLint(BLOCK_WITH_BRACES_FAIL)
+
+            assertThat(findings).hasSize(10)
+            assertThat(findings).hasTextLocations(
+                109 to 111,
+                201 to 205,
+                254 to 256,
+                328 to 330,
+                301 to 303,
+                469 to 473,
+                576 to 580,
+                608 to 610,
+                770 to 774,
+                867 to 869
+            )
+        }
+    }
+
+    @Nested
+    inner class `braces never` {
+        val subject = createSubject("never", "never")
+
+        @Test
+        fun `does not report correct block without braces`() {
+            val findings = subject.compileAndLint(BLOCK_WITHOUT_BRACES_PASS)
+
+            assertThat(findings).isEmpty()
+        }
+
+        @Test
+        fun `reports incorrect block without braces`() {
+            val findings = subject.compileAndLint(BLOCK_WITHOUT_BRACES_FAIL)
+
+            assertThat(findings).hasSize(8)
+            assertThat(findings).hasTextLocations(
```

Comment:
```suggestion
            assertThat(findings).hasSize(8)
            assertThat(findings).hasTextLocations(
```
I think the `hasSize` is redundant in these verificaitons, `hasTextLocations` does a deep equal on the arrays, so if the count differs it'll fail, and also show the missing/extra/wrong locations in the error message.